### ISSUE TITLE
Support lower case for GET odbAddress requests.

### ISSUE THREAD

**Issue by oed** [3box/3box-address-server#80]
Currently only lower case ethereum addresses works when making requests. E.g:

Doesn't work:
https://beta.3box.io/address-server/odbAddress/0x5c44E8982fa3C3239C6E3C5be2cc6663c7C9387E

Works:
https://beta.3box.io/address-server/odbAddress/0x5c44e8982fa3c3239c6e3c5be2cc6663c7c9387e

**Comment by ghiliweld** [comment]
@oed do we want to simply convert any incoming uppercase addresses to lowercase strings?

**Comment by oed** [comment]
yes

**Comment by msterle** [comment]
TLDR

Is there a reason we want to apply this transformation, rather than a lower-case validation? I think it's worth thinking about this whenever you want to transform input parameters - I'd generally suggest strict validation with informative errors instead.

-----

Some thoughts on transforming input parameters versus strict parameter validation:

This kind of transformation would be somewhat opaque, in that although we could return the actual address string in the body of POST /link and POST /odbAddresses, the GET /odbaddress doesn't return the address string being fetched in the response body. In this case, we can either change the response schema for that endpoint to include the actual address string that was set, or enforce a hidden (to the client) transformation across the board for consistency (the client would think they had set a link for the address string `0xA23...` but it was actually set for `0xa23...`).

The problem with not informing the client about a transformation is that it's possibly keeping its own records, and there is now a mismatch between these two stores. For example, let's say we set up a reverse lookup, GET /address, which accepts a DID and returns the associated address. If a client has stored the string '0xA23', linked it, and now uses this reverse lookup to test that it was correctly linked, they would get back the string '0xa23', and fail their test.

This is a minor example of this sort of problem, but APIs are interacted with by programs, and even though we recognize this string as a hex number, the only type information implicit in it is that it's a string, which are case sensitive by default.

**Comment by oed** [comment]
That's a very good point that @msterle. An informative error is better 👍
